### PR TITLE
Feature/rm 9286 asp net core hosting strategy should support non localhost hostnames and https by default

### PR DESCRIPTION
**Pull request by Frazerus**
https://re-motion.atlassian.net/browse/RM-9286

### CODE REVIEW

File: Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs [hunk 1]
```diff
@@ -28,13 +27,18 @@ public class AspNetCoreHostingStrategy : IHostingStrategy
   /// </summary>
   /// <param name="testSiteLayoutConfiguration">The configuration of the layout of the used test site.</param>
   /// <param name="port">Port to be used.</param>
-  public AspNetCoreHostingStrategy (ITestSiteLayoutConfiguration testSiteLayoutConfiguration, int port)
+  /// <param name="hostname">Hostname to be used. Default is 'localhost'.</param>
+  /// <param name="useHttps">Determines whether the site is hosted using https.</param>
+  public AspNetCoreHostingStrategy (ITestSiteLayoutConfiguration testSiteLayoutConfiguration, int port, string? hostname, bool useHttps)
```
Comment (a-ctor) [inline Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs:32]: I would make `hostname` not-null and add the `localhost` default in line 55. Having a nullable parameter here feels strange.

File: Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs [hunk 2]
```diff
@@ -20,6 +17,8 @@ public class AspNetCoreHostingStrategy : IHostingStrategy
 {
   private readonly ITestSiteLayoutConfiguration _testSiteLayoutConfiguration;
   private readonly int _port;
+  private readonly bool _useHttps;
+  private readonly string _hostname;
```
Comment (a-ctor) [inline Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs:21]:
```suggestion
  private readonly string _hostname;
  private readonly bool _useHttps;
```

:p

File: Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs [hunk 3]
```diff
@@ -63,10 +69,12 @@ public void DeployAndStartWebApplication ()
           "The executable's path was not set while trying to host with ASP.NET Core. Please add the 'processPath' attribute to the 'testSiteLayout' configuration.");
     }
 
+    var http = _useHttps ? "https" : "http";
```
Comment (a-ctor) [inline Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs:72]: We could use `protocol` as variable name here instead. That would fit `http` and `https`

File: Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs (hunk 1 above)
Comment (Frazerus) [reply Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs:32]: line 55 always either receives a hostname or throws an error, there is no nullability there. I think we should just remove the localhost default.

File: Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs [hunk 4]
```diff
@@ -28,13 +27,18 @@ public class AspNetCoreHostingStrategy : IHostingStrategy
   /// </summary>
   /// <param name="testSiteLayoutConfiguration">The configuration of the layout of the used test site.</param>
   /// <param name="port">Port to be used.</param>
-  public AspNetCoreHostingStrategy (ITestSiteLayoutConfiguration testSiteLayoutConfiguration, int port)
+  /// <param name="hostname">Hostname to be used..</param>
+  /// <param name="useHttps">Determines whether the site is hosted using https.</param>
+  public AspNetCoreHostingStrategy (ITestSiteLayoutConfiguration testSiteLayoutConfiguration, int port, string hostname, bool useHttps)
```
Comment (MichaelKetting) [inline Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs:32]: Do we have unit test for this type?

**Review by MichaelKetting — APPROVED** [review]
See comments. I don't need to re-check. @a-ctor is fine.

File: Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs (hunk 4 above)
Comment (Frazerus) [reply Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs:32]: We dont have unit tests for any hosting strategies.

File: Remotion/ObjectBinding/Web.Development.WebTesting.IntegrationTests/App.config [hunk 5]
```diff
@@ -28,7 +28,7 @@
       closeBrowserWindowsOnSetUpAndTearDown="false"
       cleanUpUnmatchedDownloadedFiles="false"
       requestErrorDetectionStrategy="AspNet">
-    <hosting name="AspNetCore" type="AspNetCore" port="60402" />
+    <hosting name="AspNetCore" type="AspNetCore" port="60402" hostname="localhost" useHttps="false"/>
```
[the remark this answers is not in the document]
Comment (Frazerus) [reply Remotion/ObjectBinding/Web.Development.WebTesting.IntegrationTests/App.config:31]: We do not.

**Pull request merged**